app/models/provider_interface/provider_applications_filter.rb

```diff
@@ -160,5 +160,24 @@ def provider_locations_filters
         }
       end
     end
+
+    def subject_filter
+      provider_ids = applied_filters[:provider] || provider_user.providers
```

Comment:
talked in person: include all subjects for providers

Reply:
The problem still persists even with provider ids. To match the list of providers returned we will need to use: `ProviderOptionsService.new(provider_user).providers` as we do in `provider_filter` to keep the list of providers aligned.